libqtile/widget/base.py

```diff
@@ -615,8 +729,10 @@ class ThreadPoolText(_TextBox):
         ),
     ]  # type: list[tuple[str, Any, str]]
 
-    def __init__(self, text, **config):
-        super().__init__(text, width=bar.CALCULATED, **config)
+    def __init__(self, text, width=bar.CALCULATED, **config):
+        if "width" in config:
+            width = config["width"]
+        super().__init__(text, width=width, **config)
```

Comment:
The width kwarg will take the value of the width key in the `config` dict already:
```py
class Widget:
    def __init__(self, width=10, **config):
        print("width", width, "config", config)


Widget()
Widget(width=0)
Widget(width=0, var=0)
```
prints:
```
width 10 config {}
width 0 config {}
width 0 config {'var': 0}
```
so i dont think we should need anything like 733-4 besides the default argument, that you have added

Reply:
Well, what you really need to show is a separate widget inheriting `Widget` without a `width` argument but I think you're still right.

Reply:
```
>>> class Widget:
...   def __init__(self, width=10, **config):
...     print(width, config)
...
>>> class GoodChild(Widget):
...   def __init__(self, **config):
...     Widget.__init__(self, **config)
...
>>> class BadChild(Widget):
...   def __init__(self, **config):
...     Widget.__init__(self, width=10, **config)
...
>>> Widget(width=15)
15 {}
<__main__.Widget object at 0x7fd22a474c70>
>>> GoodChild(width=15)
15 {}
<__main__.GoodChild object at 0x7fd22a4741c0>
>>> BadChild(width=15)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 3, in __init__
TypeError: __main__.Widget.__init__() got multiple values for keyword argument 'width'
>>>
```

Reply:
Ah, gotcha, the duplicated `width=` is only acceptable if it's in the signature, not in any calling code, where the duplicate has to be resolved. Cool. So maybe this needs to do something like `config.setdefault("width", bar.CALCULATED)`

Reply:
Don't think we even need that. We just need to stop the child classes (`ThreadPoolText` etc) from setting a default value in their call to `_TextBox.__init__`. If it's just in config then `_TextBox` should handle it ok. That's the `GoodChild` example above.

Reply:
Some other widgets have this same issue e.g. `GroupBox`. While these shouldn't have fixed widths, I'm inclined to remove the code that causes the issue.

e.g.
https://github.com/qtile/qtile/blob/5c28cbb283c4397fb001951d6f12d0c5058988ae/libqtile/widget/groupbox.py#L47-L48